File: PHP/Behavioral/Command/Example1/ConcreteCommand.php
<?php
// ConcreteCommand.php

require_once 'Command.php';
require_once 'Receiver.php';

// ConcreteCommand binds a Receiver and delegates the execution to it
class ConcreteCommand implements Command
{
    private $receiver;

    public function __construct($receiver)
    {
        $this->receiver = $receiver; // Save the receiver instance
    }

    public function execute()
    {
        echo "ConcreteCommand: Delegating the call to the Receiver...\n";
        $this->receiver->action(); // Delegate to Receiver
    }
}

File: CSharp/Behavioral/Command/Example2/Show/Show/ConcreteCommand.cs
namespace Show
{
    public class ConcreteCommand: ICommand
    {
        private readonly Receiver _receiver;
        
        public ConcreteCommand(Receiver receiver)
        {
            _receiver = receiver;
        }
        
        public void Execute()
        {
            _receiver.Action();
        }
    }
}